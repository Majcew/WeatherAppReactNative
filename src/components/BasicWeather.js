import React, {useState, useEffect} from 'react';
import moment from 'moment';

import {View, Text, StyleSheet, Image} from 'react-native';

const BasicWeather = (props) => {
  const [updateTime, setUpdateTime] = useState(props.updateTime);
  const [temperature, setTemperature] = useState(props.temperature);
  const [basicWeatherDescription, setBasicWeatherDescription] = useState(
    props.basicWeatherDescription,
  );
  const [advancedWeatherDescription, setAdvancedWeatherDescription] = useState(
    props.advancedWeatherDescription,
  );

  useEffect(() => {
    setUpdateTime(moment(props.updateTime * 1000 + 2 * 3600 * 1000).format('DD MMM HH:mm'));
    setTemperature(Math.round(props.temperature) + "\u00B0");
    setBasicWeatherDescription(props.basicWeatherDescription);
    setAdvancedWeatherDescription(props.advancedWeatherDescription);
  }, [
    props.updateTime,
    props.temperature,
    props.basicWeatherDescription,
    props.advancedWeatherDescription,
  ]);

  return (
    <View>
      <View style={styles.centerView}>
        <Text style={styles.dateTime}>{updateTime}</Text>
        <Image
          style={styles.weatherImg}
          source={require('../img/cloudy.png')}
        />
      </View>
      <View style={styles.weatherDescription}>
        <Text style={styles.weatherTemp}>{temperature}</Text>
        <View>
          <Text style={styles.weatherDescriptionText}>
            {basicWeatherDescription}
          </Text>
          <Text style={styles.weatherDescriptionTextDetails}>
            {advancedWeatherDescription}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  centerView: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 20,
  },
  dateTime: {
    color: '#8d8d8d',
    marginBottom: 30,
  },
  weatherImg: {
    width: 155,
    height: 155,
    marginTop: 8,
    marginHorizontal: 5,
    backgroundColor: 'white',
  },
  weatherDescription: {
    flexDirection: 'row',
    marginHorizontal: 20,
    
  },
  weatherTemp: {
    fontSize: 48,
    color: '#484848',
  },
  weatherDescriptionText: {
    marginTop: 0,
    fontSize: 24,
    paddingLeft:8,
  },
  weatherDescriptionTextDetails: {
    marginTop: 0,
    fontSize: 16,
    color: '#8d8d8d',
    paddingLeft:8,
  },
});

export default BasicWeather;
