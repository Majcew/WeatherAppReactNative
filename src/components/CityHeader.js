import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  TextInput,
  Keyboard,
} from 'react-native';

const CityHeader = (props) => {
  const [city, setCity] = useState(props.city);
  const onChangeText = (text) => {
    setCity(text);
  };

  const toggleInput = () => {
    props.getWeatherCityName(city);
  };

  useEffect(() => {
    setCity(props.city);
  }, [props.city]);
  return (
    <View style={styles.headerContainer}>
      <TextInput
        style={styles.inputCity}
        underlineColorAndroid="rgba(0,0,0,0)"
        placeholder="City"
        placeholderTextColor="#fff"
        value={city}
        onChangeText={(text) => onChangeText(text)}
        onSubmitEditing={toggleInput}
      />
      <TouchableOpacity style={styles.inputButton}>
        <Text style={styles.inputText}>{city}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  headerContainer: {
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  inputCity: {
    width: 250,
    fontSize: 30,
    borderRadius: 25,
    backgroundColor: '#6f77c6',
    color: '#fff',
    paddingVertical: 10,
    marginVertical: 30,
    textAlign: 'center',
  },
  inputButton: {
    backgroundColor: '#010d79',
    borderRadius: 25,
    width: 250,
    paddingVertical: 10,
    marginVertical: 20,
  },
  inputText: {
    color: '#fff',
    textAlign: 'center',
  },
});

export default CityHeader;
