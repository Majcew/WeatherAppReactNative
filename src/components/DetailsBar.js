import React, {useState, useEffect} from 'react';
import moment from 'moment';

import {
  View,
  Text,
  StyleSheet,
  Image,
} from 'react-native';

const DetailsBar = (props) => {
    const [sunriseTime, setSunriseTime] = useState(props.sunriseTime);
    const [sunsetTime, setSunsetTime] = useState(props.sunsetTime);
    const [airPressure, setAirPressure] = useState(props.airPressure);
    const [humidity, setHumidity] = useState(props.humidity);
    
    
  
    useEffect(() => { 
        setSunriseTime(moment(props.sunriseTime*1000+2*3600*1000).format('HH:mm')); //tutaj super biblioteka moment do konwejsji timestamp
        setSunsetTime(moment(props.sunsetTime*1000+2*3600*1000).format('HH:mm'));
        setAirPressure(props.airPressure);
        setHumidity(props.humidity);
      }, [props.sunriseTime,props.sunsetTime,props.airPressure,props.humidity]);
  
      
    
    return (
        <View style={styles.iconBox}>
        <View>
          <View style={styles.box1}>
            <Image
              style={styles.iconInSquare}
              source={require('../img/sunsetup.png')}
            />
          </View>
    <Text style={{textAlign: 'center'}}>{sunriseTime}</Text>
        </View>
        <View>
          <View style={styles.box2}>
            <Image
              style={styles.iconInSquare}
              source={require('../img/sunsetdown.png')}
            />
          </View>
          <Text style={{textAlign: 'center'}}>{sunsetTime}</Text>
        </View>
        <View>
          <View style={styles.box3}>
            <Image
              style={styles.iconInSquare}
              source={require('../img/hpa.png')}
            />
          </View>
          <Text style={{textAlign: 'center'}}>{airPressure}</Text>
        </View>
        <View>
          <View style={styles.box4}>
            <Image
              style={styles.iconInSquare}
              source={require('../img/humidity.png')}
            />
          </View>
          <Text style={{textAlign: 'center'}}>{humidity + " %"}</Text>
        </View>
      </View>
    );
  };

  const styles = StyleSheet.create({
    iconInSquare: {
        width: 35,
        height: 35,
        marginLeft: 8,
        marginTop: 6,
      },
      iconBox: {
        marginTop: 25,
        marginHorizontal: 5,
        flexDirection: 'row',
        justifyContent: 'space-around',
      },
      box1: {
        width: 50,
        height: 50,
        backgroundColor: 'rgb(10, 132, 255)',
        borderRadius: 4,
      },
      box2: {
        width: 50,
        height: 50,
        backgroundColor: 'rgb(48, 209, 88)',
        borderRadius: 4,
      },
      box3: {
        width: 50,
        height: 50,
        backgroundColor: 'rgb(94, 92, 230)',
        borderRadius: 4,
      },
      box4: {
        width: 50,
        height: 50,
        backgroundColor: 'rgb(255, 159, 10)',
        borderRadius: 4,
      },
  });

  export default DetailsBar;