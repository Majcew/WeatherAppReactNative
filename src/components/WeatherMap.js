const api_key = 'c4ffc65c1eda93d335ccc1b45fbab2cc';
const url_prefix = 'https://tile.openweathermap.org/map/';
const url_sufix = '/{z}/{x}/{y}.png?appid=';
const layers = [
  'https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4NXVycTA2emYycXBndHRqcmZ3N3gifQ.rJcFIG214AriISLbB6B5aw',
  url_prefix + 'temp_new' + url_sufix + api_key,
  url_prefix + 'pressure_new' + url_sufix + api_key,
  url_prefix + 'clouds_new' + url_sufix + api_key,
  url_prefix + 'precipitation_new' + url_sufix + api_key,
  url_prefix + 'wind_new' + url_sufix + api_key,
];

const html_map = (lat, lon, zoom, layer) => `
<!DOCTYPE html>
<html>
<head>
	
	<title>Quick Start - Leaflet</title>
	<meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
	
	<link rel="shortcut icon" type="image/x-icon" href="docs/images/favicon.ico" />
    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.6.0/dist/leaflet.css" integrity="sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ==" crossorigin=""/>
    <script src="https://unpkg.com/leaflet@1.6.0/dist/leaflet.js" integrity="sha512-gZwIG9x3wUXg2hdXF6+rVkLF/0Vi9U8D2Ntg4Ga5I5BZpVkVxlJWbSQtXPSiUTtC0TjtGOmxa1AJPuV0CPthew==" crossorigin=""></script>
	
</head>
<body>
<div id="mapid" style="width: 100wh; height: 100vh;"></div>
<script>
	var mymap = L.map('mapid').setView([${lat}, ${lon}], ${zoom});
	L.tileLayer('${layers[layer]}', {
        maxZoom: ${zoom},
        minZoom: ${zoom},
		attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors, ' +
			'<a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, ' +
			'Imagery Â© <a href="https://www.mapbox.com/">Mapbox</a>',
		id: 'mapbox/streets-v11',
        tileSize: 512,
        zoomOffset: -1
	}).addTo(mymap);
	L.marker([${lat}, ${lon}]).addTo(mymap)
		.bindPopup("<b>I'm here</b><br />lat:${lat}<br />lon:${lon}").openPopup();
	
	function onMapClick(e) {
		popup
			.setLatLng(e.latlng)
			.setContent("You clicked the map at " + e.latlng.toString())
			.openOn(mymap);
	}
	mymap.on('click', onMapClick);
</script>
</body>
</html>
`;

export default html_map;
