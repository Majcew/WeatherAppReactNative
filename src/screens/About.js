import React, {useCallback} from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  Image,
  StyleSheet,
  Linking,
  Alert,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const OpenURLText = ({url, children}) => {
  const handlePress = useCallback(async () => {
    const supported = await Linking.canOpenURL(url);

    if (supported) {
      await Linking.openURL(url);
    } else {
      Alert.alert(`Don't know how to open this URL: ${url}`);
    }
  }, [url]);

  return (
    <Text style={{color: 'blue'}} onPress={handlePress}>
      {children}
    </Text>
  );
};

const About = () => {
  return (
    <View style={styles.mainView}>
      <View>
        <View style={styles.icon}>
          <Icon name="umbrella" size={128} color="rgb(88, 86, 214)" />
        </View>
        <Text style={styles.appName}>WeatherAPP</Text>
      </View>
      <View style={styles.textView}>
        <Text style={styles.textHeader}>About App:</Text>
        <Text>
          Weather application presenting actual weather data with the forecast
          for a few next day.
        </Text>
      </View>
      <View style={styles.textView}>
        <Text style={styles.textHeader}>Contributors:</Text>
        <Text>Szymon Joszko</Text>
        <Text>Krzysztof Dragon</Text>
        <Text>Wiktor Hosumbek</Text>
        <Text>Szymon Babula</Text>
      </View>
      <View style={styles.textView}>
        <Text style={styles.textHeader}>Links:</Text>
        <OpenURLText url={'https://github.com/Majcew/WeatherAppReactNative'}>
          GitHub
        </OpenURLText>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  mainView: {
    flex: 1,
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 40,
  },
  icon: {
    marginLeft: 22,
  },
  appName: {
    color: 'rgb(88, 86, 214)',
    marginBottom: 40,
    fontSize: 28,
  },
  textView: {
    marginVertical: 10,
    alignItems: 'center',
  },
  textHeader: {
    fontWeight: 'bold',
  },
  drawer: {
    marginRight: 10,
    width: 30,
    height: 30,
  },
});

About.navigationOptions = ({navigation}) => ({
  headerRight: (
    <TouchableOpacity
      onPress={() => {
        navigation.openDrawer();
      }}>
      <Image style={styles.drawer} source={require('../img/menu.png')} />
    </TouchableOpacity>
  ),
});

export default About;
