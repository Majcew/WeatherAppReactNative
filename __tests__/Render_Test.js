//__tests__/Render_Test.js
import React from 'react';
import BasicWeather from '../src/components/BasicWeather';
import CityHeader from '../src/components/CityHeader';
import DetailsBar from '../src/components/DetailsBar';
//import Location from '../src/components/Location';
import WeatherMap from '../src/components/WeatherMap';
import About from '../src/screens/About';
//import Main from '../src/screens/Main';
import Map from '../src/screens/Map';
import WeatherForecast from '../src/screens/WeaterForecast';

import renderer from 'react-test-renderer';

jest.useFakeTimers();

const weather = {
  base: 'stations',
  clouds: {all: 90},
  cod: 200,
  coord: {lat: 50.3, lon: 18.68},
  dt: 1590249302,
  id: 3099230,
  main: {
    feels_like: 11.71,
    humidity: 72,
    pressure: 1019,
    temp: 14.63,
    temp_max: 15,
    temp_min: 13.89,
  },
  name: 'Gliwice',
  sys: {
    country: 'PL',
    id: 1705,
    sunrise: 1590202074,
    sunset: 1590258992,
    type: 1,
  },
  timezone: 7200,
  visibility: 10000,
  weather: [
    {description: 'overcast clouds', icon: '04d', id: 804, main: 'Clouds'},
  ],
  wind: {deg: 170, speed: 4.1},
};

describe('Rendering: ', () => {
  describe('Components: ', () => {
    describe('BasicWeather.js', () => {
      test('Result', async () => {
        const tree = renderer
          .create(
            <BasicWeather
              updateTime={weather?.dt}
              temperature={weather?.main?.temp}
              basicWeatherDescription={weather?.weather[0]?.main}
              advancedWeatherDescription={weather?.weather[0]?.description}
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('CityHeader.js', () => {
      test('Result ', () => {
        const tree = renderer.create(<CityHeader />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('DetailsBar.js', () => {
      test('Result ', async () => {
        const tree = renderer.create(<DetailsBar />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    // describe('Location.js', () => {
    //   test('Result ', async () => {
    //     const tree = renderer.create(<Location />).toJSON();
    //     expect(tree).toMatchSnapshot();
    //   });
    // });
    describe('WeatherMap.js', () => {
      test('Result ', async () => {
        const tree = renderer.create(<WeatherMap />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
  describe('Screens:', () => {
    describe('About.js', () => {
      test('Result ', async () => {
        const tree = renderer.create(<About />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    // describe('Main.js', () => {
    //   test('Result ', async () => {
    //     const tree = renderer.create(<Main />).toJSON();
    //     expect(tree).toMatchSnapshot();
    //   });
    // });
    describe('Map.js', () => {
      test('Result ', async () => {
        const tree = renderer.create(<Map />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('WeatherForecast.js', () => {
      test('Result ', async () => {
        const tree = renderer.create(<WeatherForecast />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
