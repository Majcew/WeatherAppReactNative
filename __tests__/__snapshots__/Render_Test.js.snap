// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering:  Components:  BasicWeather.js Result 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#8d8d8d",
          "marginBottom": 30,
        }
      }
    >
      1590249302
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/img/cloudy.png",
        }
      }
      style={
        Object {
          "backgroundColor": "white",
          "height": 155,
          "marginHorizontal": 5,
          "marginTop": 8,
          "width": 155,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#484848",
          "fontSize": 48,
        }
      }
    >
      14.63
    </Text>
    <View>
      <Text
        style={
          Object {
            "fontSize": 24,
            "marginTop": 0,
            "paddingLeft": 8,
          }
        }
      >
        Clouds
      </Text>
      <Text
        style={
          Object {
            "color": "#8d8d8d",
            "fontSize": 16,
            "marginTop": 0,
            "paddingLeft": 8,
          }
        }
      >
        overcast clouds
      </Text>
    </View>
  </View>
</View>
`;

exports[`Rendering:  Components:  CityHeader.js Result  1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="City"
    placeholderTextColor="black"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomWidth": 0.15,
        "color": "black",
        "fontSize": 18,
      }
    }
    underlineColorAndroid="rgba(0,0,0,0)"
  />
</View>
`;

exports[`Rendering:  Components:  DetailsBar.js Result  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginHorizontal": 5,
      "marginTop": 25,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(10, 132, 255)",
          "borderRadius": 4,
          "height": 50,
          "width": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/img/sunsetup.png",
          }
        }
        style={
          Object {
            "height": 35,
            "marginLeft": 8,
            "marginTop": 6,
            "width": 35,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(48, 209, 88)",
          "borderRadius": 4,
          "height": 50,
          "width": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/img/sunsetdown.png",
          }
        }
        style={
          Object {
            "height": 35,
            "marginLeft": 8,
            "marginTop": 6,
            "width": 35,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(94, 92, 230)",
          "borderRadius": 4,
          "height": 50,
          "width": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/img/hpa.png",
          }
        }
        style={
          Object {
            "height": 35,
            "marginLeft": 8,
            "marginTop": 6,
            "width": 35,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 159, 10)",
          "borderRadius": 4,
          "height": 50,
          "width": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/img/humidity.png",
          }
        }
        style={
          Object {
            "height": 35,
            "marginLeft": 8,
            "marginTop": 6,
            "width": 35,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      undefined %
    </Text>
  </View>
</View>
`;

exports[`Rendering:  Components:  WeatherMap.js Result  1`] = `
"
<!DOCTYPE html>
<html>
<head>
	
	<title>Quick Start - Leaflet</title>
	<meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no\\" />
	
	<link rel=\\"shortcut icon\\" type=\\"image/x-icon\\" href=\\"docs/images/favicon.ico\\" />
    <link rel=\\"stylesheet\\" href=\\"https://unpkg.com/leaflet@1.6.0/dist/leaflet.css\\" integrity=\\"sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ==\\" crossorigin=\\"\\"/>
    <script src=\\"https://unpkg.com/leaflet@1.6.0/dist/leaflet.js\\" integrity=\\"sha512-gZwIG9x3wUXg2hdXF6+rVkLF/0Vi9U8D2Ntg4Ga5I5BZpVkVxlJWbSQtXPSiUTtC0TjtGOmxa1AJPuV0CPthew==\\" crossorigin=\\"\\"></script>
	
</head>
<body>
<div id=\\"mapid\\" style=\\"width: 100wh; height: 96vh;\\"></div>
<script>
	var mymap = L.map('mapid').setView([[object Object], [object Object]], undefined);
	L.tileLayer('undefined', {
        maxZoom: undefined,
        minZoom: undefined,
		attribution: 'Map data &copy; <a href=\\"https://www.openstreetmap.org/\\">OpenStreetMap</a> contributors, ' +
			'<a href=\\"https://creativecommons.org/licenses/by-sa/2.0/\\">CC-BY-SA</a>, ' +
			'Imagery © <a href=undefined>undefined</a>',
		id: 'mapbox/streets-v11',
        tileSize: 512,
        zoomOffset: -1
	}).addTo(mymap);
	L.marker([[object Object], [object Object]]).addTo(mymap)
		.bindPopup(\\"<b>I'm here</b><br />lat:[object Object]<br />lon:[object Object]\\").openPopup();
	
	//var popup = L.popup();

	function onMapClick(e) {
		popup
			.setLatLng(e.latlng)
			.setContent(\\"You clicked the map at \\" + e.latlng.toString())
			.openOn(mymap);
	}
	mymap.on('click', onMapClick);
</script>
</body>
</html>
"
`;

exports[`Rendering:  Screens: About.js Result  1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/menu.png",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
  <Text>
    Tutaj będzie bardzo piękny tekst na temat naszej apki (nie wazne w tej chwili)
  </Text>
</View>
`;

exports[`Rendering:  Screens: Map.js Result  1`] = `
Array [
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "id": 0,
            "title": "Mapa",
          },
          Object {
            "id": 1,
            "title": "Temperatura",
          },
          Object {
            "id": 2,
            "title": "Ciśnienie",
          },
          Object {
            "id": 3,
            "title": "Chmury",
          },
          Object {
            "id": 4,
            "title": "Opady",
          },
          Object {
            "id": 5,
            "title": "Wiatr",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "dodgerblue",
                "marginHorizontal": 8,
                "marginVertical": 8,
                "padding": 15,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Mapa
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "dodgerblue",
                "marginHorizontal": 8,
                "marginVertical": 8,
                "padding": 15,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Temperatura
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "dodgerblue",
                "marginHorizontal": 8,
                "marginVertical": 8,
                "padding": 15,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Ciśnienie
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "dodgerblue",
                "marginHorizontal": 8,
                "marginVertical": 8,
                "padding": 15,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Chmury
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "dodgerblue",
                "marginHorizontal": 8,
                "marginVertical": 8,
                "padding": 15,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Opady
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "dodgerblue",
                "marginHorizontal": 8,
                "marginVertical": 8,
                "padding": 15,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Wiatr
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
  <RNCSlider
    disabled={false}
    enabled={true}
    inverted={false}
    maximumTrackTintColor="#000000"
    maximumValue={18}
    minimumTrackTintColor="#FFFFFF"
    minimumValue={1}
    onChange={[Function]}
    onRNCSliderSlidingComplete={null}
    onRNCSliderSlidingStart={null}
    onRNCSliderValueChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    step={1}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "height": 40,
          "width": "100%",
        },
      ]
    }
    thumbImage={null}
    thumbTintColor="dodgerblue"
    value={9}
  />,
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "html": "
<!DOCTYPE html>
<html>
<head>
	
	<title>Quick Start - Leaflet</title>
	<meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no\\" />
	
	<link rel=\\"shortcut icon\\" type=\\"image/x-icon\\" href=\\"docs/images/favicon.ico\\" />
    <link rel=\\"stylesheet\\" href=\\"https://unpkg.com/leaflet@1.6.0/dist/leaflet.css\\" integrity=\\"sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ==\\" crossorigin=\\"\\"/>
    <script src=\\"https://unpkg.com/leaflet@1.6.0/dist/leaflet.js\\" integrity=\\"sha512-gZwIG9x3wUXg2hdXF6+rVkLF/0Vi9U8D2Ntg4Ga5I5BZpVkVxlJWbSQtXPSiUTtC0TjtGOmxa1AJPuV0CPthew==\\" crossorigin=\\"\\"></script>
	
</head>
<body>
<div id=\\"mapid\\" style=\\"width: 100wh; height: 96vh;\\"></div>
<script>
	var mymap = L.map('mapid').setView([0, 0], 9);
	L.tileLayer('https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4NXVycTA2emYycXBndHRqcmZ3N3gifQ.rJcFIG214AriISLbB6B5aw', {
        maxZoom: 9,
        minZoom: 9,
		attribution: 'Map data &copy; <a href=\\"https://www.openstreetmap.org/\\">OpenStreetMap</a> contributors, ' +
			'<a href=\\"https://creativecommons.org/licenses/by-sa/2.0/\\">CC-BY-SA</a>, ' +
			'Imagery © <a href=Mapbox,https://www.mapbox.com/>Mapbox,https://www.mapbox.com/</a>',
		id: 'mapbox/streets-v11',
        tileSize: 512,
        zoomOffset: -1
	}).addTo(mymap);
	L.marker([0, 0]).addTo(mymap)
		.bindPopup(\\"<b>I'm here</b><br />lat:0<br />lon:0\\").openPopup();
	
	//var popup = L.popup();

	function onMapClick(e) {
		popup
			.setLatLng(e.latlng)
			.setContent(\\"You clicked the map at \\" + e.latlng.toString())
			.openOn(mymap);
	}
	mymap.on('click', onMapClick);
</script>
</body>
</html>
",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>,
]
`;

exports[`Rendering:  Screens: WeatherForecast.js Result  1`] = `
<View>
  <Text>
    Laduje się (moze byc jakis obrazek ladowania)
  </Text>
</View>
`;
